### ISSUE TITLE
No documentation

### ISSUE THREAD

**Issue by greenonline** [pvd/SerialPlot#1]
While it is clear that you've not worked on this for a while, It might be useful to include a README.md. What version of OS X? Is Xcode required and what version? What version of qt? What state is the development in? Any screenshots? Are any pre-built binaries available?

---

I downloaded and unzipped. Ran `make` in a terminal and got:

```
$ make
make: *** No rule to make target `../../../QtSDK/Desktop/Qt/473/gcc/mkspecs/macx-g++/qmake.conf', needed by `Makefile'.  Stop.
$
```

After investigating. I see that I will need to install Qt5, which seems incompatible with Xcode 9.4. Hmmm...

Running on OS X 10.13.6, Xcode 9.4.

---

FWIW, I am trying to find an OS X version of [serialplot](https://github.com/hyOzd/serialplot) and I thought your project could be useful.

**Comment by pvd** [comment]
O my god that is code from 2011 or something I totally forgot I wrote that. Maybe it is better remove this completely, I have nothing on my macbook to even try to compile it. 

FYI, the Arduino IDE has a simple serial plotter that works just fine with MacOS.

**Comment by greenonline** [comment]
No, please don't delete it. I have a number of old OS X development environments,  from  Leopard onwards, and even prior on PowerPC, so I might be able to quickly set up an environment for your code.. I'd like to try to help in the development if possible. However, I must admit that I am unfamiliar with Qt (but I'm willing to give anything a go).  Can you remember the versions of software that you used to get as far as you did with your code? For example, the OS X version, Xcode version, QT code version (I presume that it wasn't QT5)?

I really would like to be able to use a nice Serial Plotter on OS X (similar to the [serialplot](https://github.com/hyOzd/serialplot) application for Windows). The serial plotter in the Arduino IDE is just appalling for any serious work. To begin with - it just autoscales too much.

----

Ok, after about 6 hours I've almost managed to get it to compile (I'm just one step away - linking `qextserialport` is blocking me). 

I've documented the whole process (along with a process summary, required links, software, etc.) here: [Qt coding adventure](https://gr33nonline.wordpress.com/2021/07/25/serialplot-a-qt-adventure-on-os-x/).

**Comment by greenonline** [comment]
OK, I've finally made it compile and got the application running...!

I'll come up with a brief README.md in a day or two.

**Comment by greenonline** [comment]
OK, I have finally come up with a [README.md](https://github.com/greenonline/serialplot_OS_X_port/blob/main/Readme_SerialPlot.md) detailing the HOWTO. I've used your **About** text as part of the introduction.

It isn't complete as I want to add a final section on bundling the frameworks to make it portable.

However, please take a look and see what you think.

If you'd permit me, would it be OK for me to submit the readme to your repository, or you rather that you did it, in order to maintain ownership..?

**Comment by pvd** [comment]
Hi @greenonline, first of all hats off to you for getting this compiled and running.

I've scrolled through the README.md and I'm amazed by the work you've put into to this, the same can be said for you blog. I've just sent you a collaborator invite so you can add the README.md and commit any changes necessary. If you prefer to fork that is also fine by me, if you do so, please let me know so I can add a ref. to mine.

**Comment by greenonline** [comment]
Ok, thanks for the permission. All done. Wow, this month has really taken it out of me. I need a rest. 

I guess we can close this issue now. I'll let you do that.

There's sure to be some typos and mistakes in the Readme and/or the frameworks bundling script. Also, to be honest, I'm not convinced that the way that I hacked together the build process is *entirely* correct either. However, if you come across something blatantly wrong, or find a better process, clearly just update the doc/script - I'll do the same too.

I'll upload the binary release app too, at some point. It would be good to see if it runs on another system, other than my own, or just crashes and burns.

If I do any work on the application functionality itself then I'll probably fork it. However, I don't think I'll have the time as I need to get back to my IMU work that I paused in order to get this plotter running.

Thanks again, I've really learnt a lot doing this too, so that's productive as well...

All the best :-)
